samples/JobsAgent/JobsAgent.cpp

```diff
@@ -0,0 +1,291 @@
+                                                             ConfigCommon::tls_handshake_timeout_,
+                                                             ConfigCommon::tls_read_timeout_,
+                                                             ConfigCommon::tls_write_timeout_, true);
+            rc = p_network_connection->Initialize();
+
+            if (ResponseCode::SUCCESS != rc) {
+                AWS_LOG_ERROR(LOG_TAG_JOBS_AGENT,
+                              "Failed to initialize Network Connection. %s",
+                              ResponseHelper::ToString(rc).c_str());
+                rc = ResponseCode::FAILURE;
+            } else {
+                p_network_connection_ = std::dynamic_pointer_cast<NetworkConnection>(p_network_connection);
+            }
+#endif
+            return rc;
+        }
+
+        ResponseCode JobsAgent::DisconnectCallback(util::String client_id,
+                                                   std::shared_ptr<DisconnectCallbackContextData> p_app_handler_data) {
+            std::cout << "*******************************************" << std::endl
+                      << client_id << " Disconnected!" << std::endl
+                      << "*******************************************" << std::endl;
+            return ResponseCode::SUCCESS;
+        }
+
+        ResponseCode JobsAgent::ReconnectCallback(util::String client_id,
+                                                  std::shared_ptr<ReconnectCallbackContextData> p_app_handler_data,
+                                                  ResponseCode reconnect_result) {
+            std::cout << "*******************************************" << std::endl
+                      << client_id << " Reconnect Attempted. Result " << ResponseHelper::ToString(reconnect_result)
+                      << std::endl
+                      << "*******************************************" << std::endl;
+            return ResponseCode::SUCCESS;
+        }
+
+        ResponseCode JobsAgent::ResubscribeCallback(util::String client_id,
+                                                    std::shared_ptr<ResubscribeCallbackContextData> p_app_handler_data,
+                                                    ResponseCode resubscribe_result) {
+            std::cout << "*******************************************" << std::endl
+                      << client_id << " Resubscribe Attempted. Result" << ResponseHelper::ToString(resubscribe_result)
+                      << std::endl
+                      << "*******************************************" << std::endl;
+            return ResponseCode::SUCCESS;
+        }
+
+        ResponseCode JobsAgent::Subscribe() {
+            AWS_LOG_INFO(LOG_TAG_JOBS_AGENT, "Subscribe");
+
+            mqtt::Subscription::ApplicationCallbackHandlerPtr p_next_handler =
+                std::bind(&JobsAgent::NextJobCallback, this, std::placeholders::_1, std::placeholders::_2, std::placeholders::_3);
```

Comment:
Rather than using std::bind, I'd highly recommend you use lambda's instead. The compiler can do all sorts of optimizations, and it's cleaner.